### DISCUSSION TITLE
Disable fcm default notification on "background or quit" device state

### DISCUSSION THREAD

**Discussion by tomybudiman** [invertase/react-native-firebase#4622] [Q&A] [answered]
Hi! How to disable fcm default notification when the device state is on "background or quit"? Currently i'm using notifee library as foreground notification handler, but also i want to use the library to handle notification when the device state is on "background or quit".

![image](https://user-images.githubusercontent.com/20919730/100385674-eb1f9100-3055-11eb-9c68-276af995c5fe.png)

My code right now:
`messaging().setBackgroundMessageHandler(this.displayNotification);`

When my device state is on "background or quit", i received duplicate notifications, 1 from fcm & 1 from notifee library. How to disable the fcm default notification & use notifee notification library instead?

Thankyou.

**Comment by andersonaddo** [comment, marked as answer]
If you don't want the default notification to show from FCM, then you should be sending data-only fcm messages.

In other words, your fcm message shouldn't have a `notification` field.

**Reply by mansimultiqos** [reply]
Hey, 

I did the same, but still got default notification from firebase and body was like "This site has been updated in the background"

Please suggest what should I do next

**Reply by Jacks717** [reply]
Did you got solutions ? iam also facing same problem

**Comment by mansimultiqos** [comment]
Yes, I got the solution.
Fetched the notification from webpush object in payload. It worked in my
case.

[image: image.png]
